### PR TITLE
content: Add web app manifest doc

### LINKED ISSUE #1355: content (update/migrate): Add a web app manifest
**Author**: TBD
**Reviewer**: [Leave blank. web.dev team will assign a reviewer.]

**Target publish date:** 2019-11-11
- [X] Check this box if this is a hard deadline.

- [X] Check this box if you'd like your reviewer to fix grammar and punctuation issues for you.

**Process**

Check out the [web.dev writing process](https://web.dev/handbook/quick-start/).

When you're ready, make a copy of [this template](https://docs.google.com/document/d/1lgaNIEnXZf-RB8_p9RK22QEgpXJqnu77pLWVWVy4nuw/edit)
and fill in the proposal and outline sections. When you're finished, drop a link
to the doc in the 1st step below and check the box. A member of the web.dev team
will assign this ticket to themselves and work as your reviewer to help you
complete the remaining steps.

- [ ] 1. I've done my proposal <add a link to your proposal doc>
- [ ] 2. Proposal approved — _reviewer checks this_
- [ ] 3. I've added a draft to my proposal doc
- [ ] 4. Draft approved — _reviewer checks this_
- [ ] 5. I've submitted a pull request #<add PR number>
- [ ] 6. Pull request approved — _reviewer checks this_

### PR DESCRIPTION
**Pull request by petele**
Fixes #1355.

Changes proposed in this pull request:
- Migrates web app manifest doc from /add-manifest/ and enhances it with info from webfu

Note - not linked in the `11tydata.js` file, will do that in a separate PR when ready to launch this content.

### CODE REVIEW

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
+---
+layout: post
+title: Add a web app manifest
+authors:
+  - petelepage
+date: 2018-11-05
+updated: 2020-02-14
+description: |
+  The web app manifest is a simple JSON file that tells the browser about your
+  web application and how it should behave when 'installed' on the user's mobile
```
Comment (jpmedley) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]: Does 'installed' need to be quoted. This is the first time I've seen it that way.

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+  device or desktop.
+---
+
+The web app manifest is a JSON file that tells the browser about your
+Progressive Web App and how it should behave when installed on the user's
+desktop or mobile device. A typical manifest file includes information about
```
Comment (jpmedley) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]:
```suggestion
desktop or mobile device. A typical manifest file includes
```

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+the app name, icons it should use, and the URL that should be opened when the
```
Comment (jpmedley) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]:
```suggestion
the app name, the icons the app should use, and the URL that should be opened when the
```

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+app is launched.
+
+Manifest files are supported in Chrome, Edge, Firefox, UC Browser, Opera,
+the Samsung browser, and Safari partial support.
```
Comment (jpmedley) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]:
```suggestion
the Samsung browser. Safari has partial support.
```

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+#### `short_name` and/or `name` {: #name }
+
+You must provide at least the `short_name` or `name` property. If both are
+provided, `short_name` is used on the user's home screen, launcher, or other
+places where space may be limited. `name` is used when the app is installed.
+
+#### `icons` {: #icons }
+
+When a user installs your PWA, you can define a set of icons for the browser
+to use. These icons are used in places like the home screen, app launcher,
```
Comment (jpmedley) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]: Combine and short the first two sentences as so:

"You can define a set of icons for the browser to use after installation in places like..."

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+task switcher, splash screen, etc.
+
+The `icons` property is an array of image objects. Each object should
```
Comment (jpmedley) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]: Should include or must include?

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+#### `background_color` {: #background-color }
+
+The `background_color` property is used on the splash screen when the
+application is first launched on mobile.
+
+#### `display` {: #display }
+
+You can customize what browser UI is shown when your app is launched. For
+example, you can hide the address bar and browser chrome. Or games may want
+to go completely full screen.
```
Comment (jpmedley) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]: "Or games may want to go completely full screen." to "Games can even be made to launch fullscreen." 

Issues:
* Starting sentence with 'or'. 
* Anthropomorphizing the game.

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+        <td>
+          Opens the web app to look and feel like a standalone native
+          app. The app runs in its own window, separate from the browser, and
+          hides standard browser UI elements like the URL bar, etc.</td>
+      </tr>
+      <tr>
+        <td><code>minimal-ui</code></td>
+        <td>
+          This mode is similar to <code>fullscreen</code>, but provides the
+          user with some means to access a minimal set of UI elements for
```
Comment (jpmedley) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]:
```suggestion
          user a minimal set of UI elements for
```

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+
+#### `theme_color` {: #theme-color }
+
+The `theme_color` sets the color of the tool bar, and may be reflected in
+the app's preview in task switchers. The `theme_color` should match the
+`meta` theme color specified in your document head.
+
+## Add the web app manifest to your pages
+
+When you have created the manifest, add a `<link>` tag to all the pages of your
```
Comment (jpmedley) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]:
```suggestion
After creating the manifest, add a `<link>` tag to all the pages of your
```

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 10]
```diff
@@ -0,0 +1,231 @@
...
+
+This tab provides a human-readable version of many of your manifest's
+properties, and makes it easy to verify that all of the images are loading
+properly.
+
+## Splash screens on mobile {: #splash-screen }
+
+When your app first launches on mobile, it can take a moment for the browser
+to spin up, and the initial content to begin rendering. Instead of showing a
+white screen that may look to the user like the app is stall, the browser
```
Comment (jpmedley) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]:
```suggestion
white screen that may look to the user like the app is stalled, the browser
```

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 11]
```diff
@@ -0,0 +1,231 @@
...
+will show a splash screen, until the first paint.
```
Comment (jpmedley) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]:
```suggestion
will show a splash screen until the first paint.
```

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 12]
```diff
@@ -0,0 +1,231 @@
...
+
+Chrome will automatically creates the splash screen from the manifest
```
Comment (jpmedley) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]:
```suggestion
Chrome automatically creates the splash screen from the manifest
```

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 13]
```diff
@@ -0,0 +1,231 @@
...
+properties, including:
```
Comment (jpmedley) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]:
```suggestion
properties, specifically:
```

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 14]
```diff
@@ -0,0 +1,231 @@
...
+
+* `name`
+* `background_color`
+* `icons`
+
+The `background_color` should be the same color as the load page, to provide
+a smooth transition from the splash screen to your app.
+
+Chrome will choose the icon that closely matches the device resolution for the
+device. Providing a 192px and a 512px icon is sufficient for most cases, but
```
Comment (jpmedley) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]: Delete "for the device". It's redundant.

"Providing a 192px and a 512px icon..." to "Providing 192px and 512px icons..."

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 15]
```diff
@@ -0,0 +1,231 @@
...
+you can provide additional icons for pixel perfection.
+
+<div class="w-clearfix">&nbsp;</div>
+
+## Further reading
+
+There are several additional properties that can be added to the web app
+manifest, refer to the [MDN Web App Manifest documentation][mdn-manifest] for
+full details.
```
Comment (jpmedley) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]:
```suggestion details.
```

File: src/site/content/en/progressive-web-apps/add-manifest/index.md (hunk 1 above)
Comment (petele) [reply src/site/content/en/progressive-web-apps/add-manifest/index.md]: Good point - fixed.

File: src/site/content/en/progressive-web-apps/add-manifest/index.md (hunk 5 above)
Comment (petele) [reply src/site/content/en/progressive-web-apps/add-manifest/index.md]: Fixed!

File: src/site/content/en/progressive-web-apps/add-manifest/index.md (hunk 14 above)
Comment (petele) [reply src/site/content/en/progressive-web-apps/add-manifest/index.md]: Fixed

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 16]
```diff
@@ -0,0 +1,229 @@
...
+app, and is used to decide when the user has left the app. The `scope`
+controls the URL structure that encompasses all the entry and exit points in
+your web app. Your `start_url` must reside within the `scope`.
+
+{% Aside 'caution' %}
+If the user clicks a link in your app that navigates outside of the
+`scope`, the link will open and render within the existing PWA window. If
+you want the link to open in a browser tab, you must add `target="_blank"`
+to the `<a>` tag. On Android, links with `target="_blank"` will open in a
+Chrome Custom Tab.
```
Comment (kaycebasques) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]:
```suggestion
<a href="https://developer.chrome.com/multidevice/android/customtabs">Chrome Custom Tab</a>.
```

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 17]
```diff
@@ -0,0 +1,229 @@
...
+
+{% Aside 'gotchas' %}
+The request for the manifest is made **without** credentials (even if it's
+on the same domain), thus if the manifest requires credentials, you must
+include `crossorigin="use-credentials"` in the manifest tag.
+{% endAside %}
+
+## Test your manifest
+
+To verify your manifest is setup correctly, use the **Manifest** tab on the
```
Comment (kaycebasques) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]:
```suggestion
To verify your manifest is setup correctly, use the **Manifest** pane in the
```

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 18]
```diff
@@ -0,0 +1,229 @@
...
+**Application** panel of Chrome DevTools.
+
+<figure class="w-figure">
+  <img class="w-screenshot w-screenshot--filled"
+       src="lh-manifest.png"
+       alt="The application panel in Chrome Devtools with the manifest tab selected.">
+</figure>
+
+This tab provides a human-readable version of many of your manifest's
```
Comment (kaycebasques) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]:
```suggestion
This pane provides a human-readable version of many of your manifest's
```

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 19]
```diff
@@ -0,0 +1,229 @@
...
+Chrome will choose the icon that closely matches the device resolution for the
+device. Providing 192px and 512px icons is sufficient for most cases, but
+you can provide additional icons for pixel perfection.
+
+<div class="w-clearfix">&nbsp;</div>
+
+## Further reading
+
+There are several additional properties that can be added to the web app
+manifest, refer to the [MDN Web App Manifest documentation][mdn-manifest] for
```
Comment (kaycebasques) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]:
```suggestion
manifest. Refer to the [MDN Web App Manifest documentation][mdn-manifest] for more information.
```
Incomplete sentence

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 20]
```diff
@@ -0,0 +1,229 @@
...
+app, and is used to decide when the user has left the app. The `scope`
+controls the URL structure that encompasses all the entry and exit points in
+your web app. Your `start_url` must reside within the `scope`.
+
+{% Aside 'caution' %}
+If the user clicks a link in your app that navigates outside of the
+`scope`, the link will open and render within the existing PWA window. If
+you want the link to open in a browser tab, you must add `target="_blank"`
+to the `<a>` tag. On Android, links with `target="_blank"` will open in a
+<a href="https://developer.chrome.com/multidevice/android/customtabs">Chrome Custom Tab</a>.
```
Comment (kaycebasques) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]:
```suggestion
[Chrome Custom Tab](https://developer.chrome.com/multidevice/android/customtabs).
```

File: src/site/content/en/progressive-web-apps/add-manifest/index.md [hunk 21]
```diff
@@ -0,0 +1,229 @@
...
+  device or desktop.
+---
+
+The web app manifest is a JSON file that tells the browser about your
+Progressive Web App and how it should behave when installed on the user's
+desktop or mobile device. A typical manifest file includes
+the app name, the icons the app should use, and the URL that should be opened when the
+app is launched.
+
+Manifest files are [supported](https://caniuse.com/#feat=web-app-manifest) in Chrome, Edge, Firefox, UC Browser, Opera,
```
Comment (kaycebasques) [inline src/site/content/en/progressive-web-apps/add-manifest/index.md]:
```suggestion
Manifest files are [supported](https://developer.mozilla.org/en-US/docs/Web/Manifest#Browser_compatibility) in Chrome, Edge, Firefox, UC Browser, Opera,
```

**Pull request merged**